### PR TITLE
ENH: add compareImage to viewer

### PR DESCRIPTION
**Pull request by PaulHax**
Use setCompareImage in ThinPlateSpline example

TODO
- [x] Update itk-vtk-viewer version with this PR: https://github.com/Kitware/itk-vtk-viewer/pull/660
- [x] bootstrap UI for checkerboard compare

### CODE REVIEW

File: itkwidgets/_type_aliases.py [hunk 1]
```diff
@@ -33,3 +33,5 @@
     import xarray
     Image = Union[Image, xarray.DataArray, xarray.Dataset]
     PointSet = Union[PointSet, xarray.DataArray, xarray.Dataset]
+
+Image_Param = Union[str, Image]
```
Comment (thewtex) [inline itkwidgets/_type_aliases.py:37]: Let's just use `Union[str, Image]` so it is more directly annotated

File: itkwidgets/viewer.py [hunk 2]
```diff
@@ -230,11 +230,11 @@ def set_axes_enabled(self, enabled: bool):
     def set_background_color(self, bgColor: List[float]):
         self.queue_request('setBackgroundColor', bgColor)
 
-    def set_image(self, image: Image):
+    def set_image(self, image: Image, name: str = 'image'):
```
Comment (thewtex) [inline itkwidgets/viewer.py:233]: `'image'` -> `'Image'` to match the default

File: itkwidgets/viewer.py [hunk 3]
```diff
@@ -275,6 +275,21 @@ def set_image_volume_sample_distance(self, distance: float):
     def set_image_volume_scattering_blend(self, scattering_blend: float):
         self.queue_request('setImageVolumeScatteringBlend', scattering_blend)
 
+    def compare_images(self, fixed_image: Image_Param = 'Fixed', moving_image: Image_Param = 'Image', method: str = 'checkerboard', pattern: tuple[int, int, int] = [4, 4, 4], swap_image_order: bool = False):
```
Comment (thewtex) [inline itkwidgets/viewer.py:278]:
```
from typing import Tuple
[...]

Tuple[int, int, int]
```

`Image_Param` -> `Union[str, Image]`

I think we should emit default values for fixed_image, moving_image.

File: itkwidgets/viewer.py [hunk 4]
```diff
@@ -275,6 +275,21 @@ def set_image_volume_sample_distance(self, distance: float):
...
+        # image args may be image name or image object
+        fixed_name = 'Fixed'
+        if(isinstance(fixed_image, str)):
```
Comment (thewtex) [inline itkwidgets/viewer.py:281]: `if isinstance(fixed_image, str)`

**Review by thewtex — CHANGES_REQUESTED** [review]
Hurrah!

A few comments inline taht should also address CI

File: itkwidgets/viewer.py (hunk 3 above)
Comment (PaulHax) [reply itkwidgets/viewer.py]: Yes, removed default values for fixed_image and moving_image.

File: itkwidgets/viewer.py (hunk 2 above)
Comment (PaulHax) [reply itkwidgets/viewer.py]: Done

**Comment by PaulHax** [comment]
<p>Yessss!  Cleaner.</p>

---

 <a href='https://app.reviewnb.com/InsightSoftwareConsortium/itkwidgets/pull/640/discussion/#comment-1489460206'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

File: itkwidgets/viewer_config.py [hunk 5]
```diff
@@ -1,5 +1,5 @@
 ITK_VIEWER_SRC = (
-    "https://bafybeihnbqhcjcs5qjobubb2fmvt422jydxam4vzzdlnnh34o2qk7ojmle.on.fleek.co/"
+    "https://bafybeialn3hggruu4jvkfsxwzpainimxidicavisey4flsmc7lvo4srffy.on.fleek.co/"
 )
-PYDATA_SPHINX_HREF = "https://cdn.jsdelivr.net/npm/itk-viewer-bootstrap-ui@0.18.0/dist/bootstrapUIMachineOptions.js.es.js"
+PYDATA_SPHINX_HREF = "https://cdn.jsdelivr.net/npm/itk-viewer-bootstrap-ui@0.20.0/dist/bootstrapUIMachineOptions.js.es.js"
```
Comment (thewtex) [inline itkwidgets/viewer_config.py:4]: @bnmajor heads up, updating itk-vtk-viewer, pydata-sphinx ui here

**Review by thewtex — APPROVED** [review]
:tada: :taco:

onwards to cyan, magenta, etc.

**Pull request merged**